changes/changes.rst.j2

```diff
@@ -0,0 +1,77 @@
+{#
+ # Jinja2 template for towncrier to generate the change log from change fragments.
+ #
+ # Input variables (as of towncrier 22.8.0):
+ # - render_title (bool): Indicates whether a title should be rendered by the
+ #   template. If False, the title is being added by towncrier to the result of
+ #   the template.
+ # - sections: The change fragments, as a dict with key: section
+ #   name, value: dict with key: category (=type) name, value: dict with key:
+ #   change fragment text, value: list of issues.
+ # - definitions: The definitions of change categories (=types), as a dict
+ #   with key: type name, value: dict with items 'name', 'showcontent'.
+ # - underlines (str): The underline characters to use for the different heading
+ #   levels (following the top heading level)
+ # - versiondata (dict): Project name and version data, as a dict with items
+ #   'name', 'version', 'date'.
+ # - top_underline (str): The underline character to be used for the title
+ #   of the change log when the template generates the title.
+ # - get_indent (func): Function to get the indentation for subsequent lines
+ #   given the first line of a multi-line change fragment text.
+ #}
+{% if render_title %}
+{%   if versiondata.name %}
+{{ versiondata.name }} {{ versiondata.version }} ({{ versiondata.date }})
+{{ top_underline * ((versiondata.name + versiondata.version + versiondata.date)|length + 4)}}
+{%   else %}
+{{ versiondata.version }} ({{ versiondata.date }})
+{{ top_underline * ((versiondata.version + versiondata.date)|length + 3)}}
+{%   endif %}
+{% endif %}
+
+Released: {{ versiondata.date }}
+
+{% for section, _ in sections.items() %}
```

Comment:
I assume all those iterations are based on sorted values, so we get the same order all the time?

Reply:
The hierarchy is: sections - categories - items.
We only have one section (and I don't know what would cause to have more than one).
The categories are our headings such as "Bug fixes" and their order is defined in towncrier.toml.
The items are the change entries within the categories. towncrier orders them by issue number (items with no issue come first, their order within that group is not clear).